src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -624,14 +625,55 @@ private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
                 return typeDeclaration.Identifier.ToString() + typeDeclaration.TypeParameterList;
             }
 
-            private (string? loggerField, bool multipleLoggerFields) FindLoggerField(SemanticModel sm, TypeDeclarationSyntax classDec, ITypeSymbol loggerSymbol)
+            private (string? loggerField, bool multipleLoggerFields) FindLoggerField(SemanticModel sm, TypeDeclarationSyntax classDec,
+                Dictionary<SyntaxTree, SemanticModel> semanticModels, ITypeSymbol loggerSymbol)
             {
                 string? loggerField = null;
 
                 INamedTypeSymbol? classType = sm.GetDeclaredSymbol(classDec, _cancellationToken);
 
                 bool onMostDerivedType = true;
 
+#if ROSLYN4_8_OR_GREATER
+                IEnumerable<ClassDeclarationSyntax> primaryConstructors = classType.InstanceConstructors
```

Comment:
Unfortunately, there seems to be no way to identify primary constructor parameters in the semantic model, and the partial class declaration that contains the primary constructor may not contain a method annotated with `[LoggerMessage]`, thereby not being picked up by the `SyntaxProvider` in `LoggerMessageGenerator.Roslyn4.0.cs`, so we have to work backwards from here, potentially parsing and creating semantic models for class declarations that weren't otherwise visited.

Reply:
@CyrusNajmabadi @eiriktsarpalis any suggestion about the best way to handle that?

Reply:
@333fred mentioned in the Discord that this could be potentially be done with `LookupSymbols` or a speculative model, but:
1. `LookupSymbols` doesn't return primary constructor parameters for any positions I know will exist in all cases, even in minimal class declarations with just one partial method declaration.
2. `TryGetSpeculativeSemanticModelForMethodBody` requires a position parameter _within an existing method body_ which, once again, may not exist. https://github.com/dotnet/roslyn/issues/61564 suggests speculative models don't work in this case.